pkg/api/v1beta1/conversion.go

```diff
@@ -32,10 +32,12 @@ func init() {
 	newer.Scheme.AddStructFieldConversion(TypeMeta{}, "TypeMeta", newer.TypeMeta{}, "TypeMeta")
 	newer.Scheme.AddStructFieldConversion(TypeMeta{}, "TypeMeta", newer.ObjectMeta{}, "ObjectMeta")
 	newer.Scheme.AddStructFieldConversion(TypeMeta{}, "TypeMeta", newer.ListMeta{}, "ListMeta")
+	newer.Scheme.AddStructFieldConversion(Endpoints{}, "Endpoints", newer.Endpoints{}, "Endpoints")
```

Comment:
You should not need this if I understand - same-type, same-field.  Or do I just misunderstand what this about @lavalamp